### PR TITLE
feat(constants): Version Scheme Constants

### PR DESCRIPTION
**Pull request by souravdasslg**
1. Add version schemes to version-schemes.ts
2. Update all reference

#5088

### CODE REVIEW

File: lib/constants/version-schemes.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const VERSION_SCHEME_SEMVER = 'semver';
+export const VERSION_SCHEME_NPM = 'npm';
+export const VERSION_SCHEME_NODE = 'node';
+export const VERSION_SCHEME_HSCORP = 'hashicorp';
```
Comment (rarkins) [inline lib/constants/version-schemes.ts]:
```suggestion
export const VERSION_SCHEME_HASHICORP = 'hashicorp';
```

File: lib/config/definitions.ts [hunk 2]
```diff
@@ -593,26 +613,26 @@ const options: RenovateOptions[] = [
...
+      VERSION_SCHEME_NODE,
+      VERSION_SCHEME_NPM,
+      VERSION_SCHEME_NUGET,
+      VERSION_SCHEME_PEP440,
+      VERSION_SCHEME_POETRY,
+      VERSION_SCHEME_REGEX,
+      VERSION_SCHEME_RUBY,
+      VERSION_SCHEME_SEMVER,
+      VERSION_SCHEME_SWIFT,
     ],
```
Comment (rarkins) [inline lib/config/definitions.ts]: We should think about (later) if we can automatically import this from the constants

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/definitions.ts (hunk 2 above)
Comment (souravdasslg) [reply lib/config/definitions.ts]: I already have an idea about that, but it was out of scope for this PR, so skipped that.

**Review by rarkins — CHANGES_REQUESTED** [review]
Resolve conflicts

**Review by JamieMagee — DISMISSED** [review]

**Review by rarkins — CHANGES_REQUESTED** [review]
Resolve conflicts

**Review by rarkins — APPROVED** [review]

**Pull request merged**